numpy/_typing/_scalars.py

```diff
@@ -10,13 +11,13 @@
 # The 6 `<X>Like_co` type-aliases below represent all scalars that can be
 # coerced into `<X>` (with the casting rule `same_kind`)
 _BoolLike_co = Union[bool, np.bool_]
-_UIntLike_co = Union[_BoolLike_co, np.unsignedinteger]
-_IntLike_co = Union[_BoolLike_co, int, np.integer]
-_FloatLike_co = Union[_IntLike_co, float, np.floating]
-_ComplexLike_co = Union[_FloatLike_co, complex, np.complexfloating]
+_UIntLike_co = Union[_BoolLike_co, np.unsignedinteger[Any]]
+_IntLike_co = Union[_BoolLike_co, int, np.integer[Any]]
+_FloatLike_co = Union[_IntLike_co, float, np.floating[Any]]
+_ComplexLike_co = Union[_FloatLike_co, complex, np.complexfloating[Any, Any]]
```

Comment:
I am a bit confused why complexfloating has two parameters and we don't enforce a single one.

Reply:
In short: to represent both the real and imaginary component, _e.g._ `complex128` is currently defined as `complexfloating[_64Bit, _64Bit]`.

As for why all bit sizes here are set to `Any`? All these unions are bitsize-agnostic, so intended for situations wherein you don't know or don't care about the bitsize .

Reply:
Aparently is needed because of real and imaginary part.

The definition has the following comment.
```python
# The main reason for `complexfloating` having two typevars is cosmetic.
# It is used to clarify why `complex128`s precision is `_64Bit`, the latter
# describing the two 64 bit floats representing its real and imaginary component

class complexfloating(inexact[_NBit1], Generic[_NBit1, _NBit2]):
    ...
```

Reply:
Right, in truth it's not so much about having *both* the real and imaginary separately as it is for clarity-related reasons. Now, parametrizing w.r.t. the precision of the underlying floats is exceedingly useful, especially for accessing the real/imaginary component and mixed complex scalar/float array operations. 

But this also means we could just as easily defined, _e.g._, `complex128` as such, with the bitsize thus representing both the real and imaginary component:
``` python
complex128 = complexfloating[_64Bit]
```

Except that we're now in the somewhat non-intuitive situation wherein a 128bit-type is defined with the help of a single 64-bit parameter, hence the choice the opt for two 64-bit parameters after all.